### PR TITLE
script/ptl-tool: page through github response

### PR DESCRIPTION
**Pull request by batrick**
This fixes the script to go through the response pages from GitHub.
Previously it would only look at the first page and potentially miss
some reviews/comments.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/script/ptl-tool.py [hunk 1]
```diff
@@ -162,24 +166,57 @@
...
+            new_params = dict(params)
+            new_params.update({'page': page})
+            response = session.get(url, auth=(USER, PASSWORD), params=new_params)
+            log.debug(f"Response = {response}; links = {response.headers.get('link', '')}")
+            if response.status_code != 200:
+                log.error(f"Failed to fetch {url}: {response}")
+                sys.exit(1)
+            yield response.json()
+            link = response.headers.get('link', None)
+            page = page+1
```
Comment (tchaikov) [inline src/script/ptl-tool.py:196]: nit, add spaces around `+`. or just put
```py
page += 1
```

File: src/script/ptl-tool.py [hunk 2]
```diff
@@ -286,10 +322,8 @@ def build_branch(args):
             sys.exit(1)
         tip = fi[0].ref.commit
 
-        pr_req = requests.get("https://api.github.com/repos/ceph/ceph/pulls/{pr}".format(pr=pr), auth=(USER, PASSWORD))
-        if pr_req.status_code != 200:
-            log.error("PR '{pr}' not found: {c}".format(pr=pr,c=pr_req))
-            sys.exit(1)
+        endpoint = f"https://api.github.com/repos/{BASE_PROJECT}/{BASE_REPO}/pulls/{pr}"
+        response = list(get(session, endpoint, paging=False))[0]
```
Comment (tchaikov) [inline src/script/ptl-tool.py:326]:
```suggestion
        response = next(get(session, endpoint, paging=False))
```

File: src/script/ptl-tool.py [hunk 3]
```diff
@@ -120,6 +120,10 @@
 log = logging.getLogger(__name__)
 log.addHandler(logging.StreamHandler())
 log.setLevel(logging.INFO)
+#log.setLevel(logging.DEBUG)
+#requests_log = logging.getLogger("requests.packages.urllib3")
+#requests_log.setLevel(logging.DEBUG)
+#requests_log.propagate = True
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/script/ptl-tool.py]: Done, thanks!

File: src/script/ptl-tool.py (hunk 1 above)
Comment (batrick) [reply src/script/ptl-tool.py]: Done, thanks!

File: src/script/ptl-tool.py (hunk 2 above)
Comment (batrick) [reply src/script/ptl-tool.py]: Done, thanks!

**Pull request merged**